pkg/readiness/ready_tracker.go

```diff
@@ -534,11 +559,12 @@ func (t *Tracker) collectInvalidExpectations(ctx context.Context) {
 }
 
 func (t *Tracker) trackAssignMetadata(ctx context.Context) error {
+	hadError := false
 	defer func() {
-		t.assignMetadata.ExpectationsDone()
-		log.V(logging.DebugLevel).Info("AssignMetadata expectations populated")
-
-		_ = t.constraintTrackers.Wait()
+		if *ignoreFailedListing || !hadError {
```

Comment:
Let's add a comment that says why we care. Something like:

`if we are ignoring errors, we need to set expectations to done to prevent readiness tracker never being satisfied`

here and everywhere a similar test is made